### ISSUE TITLE
Support for SVG images

### ISSUE THREAD

**Issue by Krelborn** [gitbrent/PptxGenJS#401]
Recent versions of office support embedding of SVG images in documents. Are there any plans to support SVG in addImage?

I've done some initial investigation to see if there was an easy way to make this work, but it looks more complicated than I'd hoped.

**Comment by gitbrent** [comment]
Hi @Krelborn 

Well, it took long enough, but it does look as though SVG support is finally available in PowerPoint (current/2016 desktop and office.com).

I've added this an an enhancement.  Thanks for the request.

**Comment by NeomMob** [comment]
It would be awesome. How can we help you on that?

**Comment by gitbrent** [comment]
Update: Support for SVG images now exists for web client users

```
var pptx = new PptxGenJS();
var slide = pptx.addNewSlide();
slide.addImage({
    path:'https://upload.wikimedia.org/wikipedia/commons/2/28/Cadenas-ferme-vert.svg',
    x:1, y:1, w:3.0, h:3.0
});
pptx.save('PptxGenJS-SVG');
```

![screen shot 2018-09-23 at 21 26 23](https://user-images.githubusercontent.com/7218970/45936036-7a612380-bf77-11e8-922e-977dea017e99.png)

**Comment by gitbrent** [comment]
I have not tested the use of `data` as of yet; I was also hoping to add support for passing the SVG XML as well.

**Comment by NeomMob** [comment]
From SVG xml would be awesome! Thanks for your work.

**Comment by gitbrent** [comment]
**Update**:
SVG images can now be created using either `data` or `path`.

I've added a demo for both cases in the demo.

![screen shot 2018-09-27 at 19 07 13](https://user-images.githubusercontent.com/7218970/46181036-52c8ce80-c289-11e8-806f-2806ca47b454.png)

**Comment by gitbrent** [comment]
**NOTE**
SVG images are only supported in web browsers and browser-based apps (Angular, etc.) as `<canvas>` is required

**Comment by clement-igonet** [comment]
Issue closed but nodejs (server) version is not implemented.
Could we reopen until support or split thr issue in two (browser and server) ?

**Comment by gitbrent** [comment]
Hi @clement-igonet 

The only way to provide support for SVG on Node is to include an entire browser engine as a module - as the conversion used is Canvas-based - so I've opted to not include that feature on Node.

**Comment by gitbrent** [comment]
SVG images can be generated in Node again! See Issue #515 and Pull #528 

Coming in Version 3.0

**Comment by Sher-V** [comment]
Using path like:  'data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" width="1564" height="700"><foreignObject...' I get empty slide 
![image](https://user-images.githubusercontent.com/43819353/106185764-4e43b600-61b4-11eb-8925-5a743449aee7.png)

**Comment by gitbrent** [comment]
@Sher-V - i'm guessing the `<` and `>` characters are messing with the XML. The library doesn't make any attempts to escape such characters in data strings.

**Comment by hrithikgautham** [comment]
i tried adding b64 format...
pptSlide.addImage({path: "data:image/svg+xml;base64,"+component.content, w: "100%", h: "100%"});
where, component.content is SVG in b64 format. It is making a get request. and the response of the get request is b64 decoded version of component.content. but the image not appearing in ppt. it says "PowerPoint found a problem with content", and when i open the ppt, instead of image, it shows "the picture cant be displayed".

**Comment by chrisv-dev** [comment]
I experienced an issue while opening a generated pptx with `LibreOffice Impress`. It looks like adding a SVG image doesn't work well in this case (at least when using `path`).
When the image is scaled up, it becomes blurry (see image).

![libreoffice](https://user-images.githubusercontent.com/15820871/151410164-6746e445-f31c-4099-8b73-564acd484979.png)

Using the same svg (local file) and adding it to a slide (drag and drop `.svg` into LibreOffice Impress) is working fine:
![libreoffice_manual](https://user-images.githubusercontent.com/15820871/151409990-c591f1d1-67b0-4c6e-b6c9-ed7997ea0b14.png)



Tested with: LibreOffice Impress, version 7.1.8.1
Example: https://codesandbox.io/s/goofy-carson-5pzyl
